static/get_grpcurl

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
+
+set -e
+
+detect_os () {
+  RAW_OS=$(uname -s)
+  OS="linux"
+  case "$RAW_OS" in
+    "Linux") OS="linux";;
+    "Darwin") OS="osx";;
+    *) echo "Unrecognized operating system \"${RAW_OS}\"" 1>&2; exit 1;;
+  esac
+  echo "${OS}"
+}
+
+detect_architecture () {
+  RAW_ARCH=$(uname -m)
+  ARCH="x86_64"
+  case "$RAW_ARCH" in
+    "x86_64") ARCH="x86_64";;
+    "i386") ARCH="x86_32";;
+    *) echo "Unrecognized architecture \"${RAW_ARCH}\"" 1>&2; exit 1;;
+  esac
+  echo "${ARCH}"
+}
+
+detect_platform () {
+  OS=$(detect_os)
+  ARCH=$(detect_architecture)
+  echo "${OS}_${ARCH}"
+}
+
+confirm () {
+  echo "$1"
+  read -p "[y/n]" -r RESPONSE </dev/tty
+  printf "\n"
+  case "$RESPONSE" in
+    "yes" | "y" | "YES" | "Y" ) return 0;;
+    "no" | "n" | "NO" | "N" ) echo "Aborting." 1>&2; exit 1;;
+    *) echo "Unrecognized response. Aborting." 1>&2; exit 1;;
+  esac
+}
+
+ensure_command () {
+  if command -v "$1" 1>/dev/null 2>&1; then
+    return 0
+  else
+    echo "$1 is not installed. Please install it to proceed." 1>&2
+    exit 1
+  fi
+}
+
+GITHUB_URL="https://github.com"
+GRPCURL_REPO="fullstorydev/grpcurl"
+
+PLATFORM=$(detect_platform)
+
+BIN_DIR="${HOME}/.grpcurl/bin"
+
+confirm "This script will perform a user install of grpcurl. Would you like to proceed?"
+
+ensure_command "curl"
```

Comment:
You also technically need to ensure cut, grep, tar, mkdir and cp.

Reply:
Fair enough. Shouldn't there be some sort of "GNU collection standard"? I know there's a POSIX shell standard, but it seems like there should also be some lowest common denominator that busybox et al. can conform to.

Reply:
GNU's coreutils will usually be always installed, that's true, and has cut, mkdir and cp. Tar and grep... I've seen oddballs distros without them by default.

Might as well be thorough anyway, it's not costing anything :-)